test/uk/gov/hmrc/excisemovementcontrolsystemapi/connector/EISSubmissionConnectorSpec.scala

```diff
@@ -95,7 +115,7 @@ class EISSubmissionConnectorSpec extends PlaySpec with BeforeAndAfterEach with E
     }
 
     "send a request with the right parameters" in {
-      val encodeMessage = encoder.encodeToString(xml.getBytes(StandardCharsets.UTF_8))
+      val encodeMessage = encoder.encodeToString(controlWrappedXml.toString.getBytes(StandardCharsets.UTF_8))
```

Comment:
if you address the comment above about the encoding function in EmcsUtils this may change an you my need to set up the mock.

Reply:
It doesn't feel right to mock the encoding logic here as it should just be used as-is. In the test I'm now creating an expected encoded message and also mocking the return from the encode message to be the same as the expected encoded message. This seems like we are mocking something that shouldn't be mocked and the encode method should be called as is but we have an expected encoded message in the test and it just checks if its being encoded properly rather that it being encoded as you've told it to in the mock.